docs/snakefiles/rules.rst

```diff
@@ -713,6 +699,98 @@ In the R Markdown file you can insert output from a R command, and access variab
 A link to the R Markdown document with the snakemake object can be inserted. Therefore a variable called ``rmd`` needs to be added to the ``params`` section in the header of the ``report.Rmd`` file. The generated R Markdown file with snakemake object will be saved in the file specified in this ``rmd`` variable. This file can be embedded into the HTML document using base64 encoding and a link can be inserted as shown in the example above.
 Also other input and output files can be embedded in this way to make a portable report. Note that the above method with a data URI only works for small files. An experimental technology to embed larger files is using Javascript Blob `object <https://developer.mozilla.org/en-US/docs/Web/API/Blob>`_.
 
+Julia_
+~~~~~~
+
+.. _Julia: https://julialang.org
+
+.. code-block:: python
+
+    rule NAME:
+        input:
+            "path/to/inputfile",
+            "path/to/other/inputfile"
+        output:
+            "path/to/outputfile",
+            "path/to/another/outputfile"
+        script:
+            "path/to/script.jl"
+
+In the Julia_ script, a ``snakemake`` object is available, which can be accessed similar to the :ref:`Python case <Python>`, with the only difference that you have to index from 1 instead of 0.
+
+Rust_
+~~~~~
+
+.. _Rust: https://www.rust-lang.org/
+
+.. code-block:: python
+
+    rule NAME:
+        input:
+            "path/to/inputfile",
+            "path/to/other/inputfile"
+        output:
+            "path/to/outputfile",
+            "path/to/another/outputfile"
+        params:
+            seed=4
+        script:
+            "path/to/script.rs"
+
+The ability to execute Rust scripts is facilitated by |rust-script|_. As such, the
+script must be a valid ``rust-script`` script.
+
+In the Rust script, an instance of a ``Snakemake`` struct can be obtained via
+
+.. code-block:: rust
+
+    let snakemake = Snakemake::load()?;
```

Comment:
Why is this needed? Can't it be done in the preamble?

Reply:
You can have scripts both with and without an explicit main fn. In the latter case, it sure could be done in the preamble, in the former case it's not that easy in general. There probably is a nice way to solve this, I'm just not sure yet what that'd look like.
*If* we ditched serde (be it -pickle or -json) and provided our own conversion to some custom types (purely on the snakemake side), we could probably make the snakemake instance be constant / a singleton / whatever.

Reply:
But you should be able to use lazy_static for this: https://docs.rs/lazy_static
Should work in both cases.